lib/resty/core/base.lua

```diff
@@ -232,4 +232,23 @@ _M.FFI_DONE = -4
 _M.FFI_DECLINED = -5
 
 
+local exdata
```

Comment:
Might be worth encapsulating `exdata` in a `do ... end` block here, to prevent future contributors to this file from calling/tampering with it in subsequently implemented functions. As in:
```lua
do
    local exdata

    local pok, exdata = pcall(...)
    if pok and type(exdata) == "function" then
        function _M.get_request()
            local r = exdata() -- only allowed caller of this function, which must not be tampered with
            -- ...
        end

    else
        -- ...
    end
end
```